### PR TITLE
added mmrm docs

### PR DESCRIPTION
**Pull request by gowerc**
Initial attempt at mmrm help files

### CODE REVIEW

File: R/mmrm.R [hunk 1]
```diff
@@ -1,17 +1,33 @@
 
 
 
-#' Title
+#' Construct random effects formula
 #'
-#' @param cov_struct TODO
-#' @param groups TODO
+#' Constructs a character representation of the random effects formula
+#' for fitting a MMRM for subject by visit in the format required for glmmTMB.
```
Comment (nociale) [inline R/mmrm.R:7]:
```suggestion
#' for fitting MMRM in the format required for glmmTMB.
```

File: R/mmrm.R [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+#'
+#' assuming the user specified a covariance structure of "us" and that no groups
```
Comment (nociale) [inline R/mmrm.R:9]:
```suggestion
#' For example assuming the user specified a covariance structure of "us" and that no groups
```

File: R/mmrm.R [hunk 3]
```diff
@@ -1,17 +1,33 @@
...
+#' were provided this will return
+#'
+#' ```
+#' us(0 + visit | subjid)
+#' ```
+#'
+#' If groups is provided then this indicates that separate covariance matrices
```
Comment (nociale) [inline R/mmrm.R:16]:
```suggestion
#' If `group` is provided then this indicates that separate covariance matrices
```

File: R/mmrm.R [hunk 4]
```diff
@@ -20,21 +36,26 @@ random_effects_expr <- function(
...
 
 
-#' TODO
+#' Creates a "MMRM" ready dataset
+#'
+#' Converts a design matrix + key variables into a command format
+#' In particular this function does the following:
+#' - Renames all covariates as `V1`, `V2`, etc to avoid issues of special characters in variable names
+#' - Ensures all key variables are of the right type
+#' - inserts the outcome, visit and subjid variables into the `data.frame`
```
Comment (nociale) [inline R/mmrm.R:45]:
```suggestion
#' - Inserts the outcome, visit and subjid variables into the `data.frame`
```

File: R/mmrm.R [hunk 5]
```diff
@@ -123,20 +156,36 @@ extract_params <- function(fit) {
 
 
 
-#' Title
+#' Fit a mmrm Model
```
Comment (nociale) [inline R/mmrm.R:159]:
```suggestion
#' Fit a MMRM model
```

File: R/mmrm.R [hunk 6]
```diff
@@ -123,20 +156,36 @@ extract_params <- function(fit) {
...
+#'
+#' @description
+#' Fits a mmrm model allowing for different covariance structures using [glmmTMB::glmmTMB()].
```
Comment (nociale) [inline R/mmrm.R:162]:
```suggestion
#' Fits a MMRM model allowing for different covariance structures using [glmmTMB::glmmTMB()].
```

File: R/mmrm.R [hunk 7]
```diff
@@ -123,20 +156,36 @@ extract_params <- function(fit) {
...
-#' @param outcome TODO
-#' @param subjid TODO
-#' @param visit TODO
-#' @param group TODO
-#' @param cov_struct TODO
-#' @param REML TODO
-#' @param same_cov TODO
-#' @param initial_values TODO
-#' @param optimizer TODO
+#' @param designmat a `data.frame` or `matrix` containing the covariates to use in the mmrm model.
```
Comment (nociale) [inline R/mmrm.R:167]:
```suggestion
#' @param designmat a `data.frame` or `matrix` containing the covariates to use in the MMRM model.
```

File: R/mmrm.R [hunk 8]
```diff
@@ -123,20 +156,36 @@ extract_params <- function(fit) {
...
+#' dummy variables must already be expanded out I.e. via [stats::model.matrix()]. Cannot contain
```
Comment (nociale) [inline R/mmrm.R:168]:
```suggestion
#' Dummy variables must already be expanded out, i.e. via [stats::model.matrix()]. Cannot contain
```

File: R/mmrm.R [hunk 9]
```diff
@@ -123,20 +156,36 @@ extract_params <- function(fit) {
...
+#' any missing values
+#' @param outcome a numeric vector. The outcome value to be reggressed on in the mmrm model.
```
Comment (nociale) [inline R/mmrm.R:170]:
```suggestion
#' @param outcome a numeric vector. The outcome value to be regressed on in the MMRM model.
```

File: R/mmrm.R [hunk 10]
```diff
@@ -123,20 +156,36 @@ extract_params <- function(fit) {
...
+#' @param subjid a character / factor vector. The subject identify use to link separate visits that belong to
```
Comment (nociale) [inline R/mmrm.R:171]:
```suggestion
#' @param subjid a character / factor vector. The subject identifier used to link separate visits that belong to
```

File: R/mmrm.R [hunk 11]
```diff
@@ -123,20 +156,36 @@ extract_params <- function(fit) {
...
+#' the same subject.
+#' @param visit a character / factor vector. Indicates which visit the outcome value occoured on.
+#' @param group a character / factor vector. Used to indicate which treatment group the patient belongs to.
```
Comment (nociale) [inline R/mmrm.R:174]:
```suggestion
#' @param group a character / factor vector. Indicates which treatment group the patient belongs to.
```

File: R/mmrm.R [hunk 12]
```diff
@@ -123,20 +156,36 @@ extract_params <- function(fit) {
...
+#' @param cov_struct a character value. Specifies which covariance structure to use. Must be one of
+#' `"us"`, `"toep"`, `"cs"` or  `"ar1"`
+#' @param REML local. Specifies whether restricted maximium likelihood should be used
```
Comment (nociale) [inline R/mmrm.R:177]:
```suggestion
#' @param REML logical. Specifies whether restricted maximium likelihood should be used
```

File: R/mmrm.R [hunk 13]
```diff
@@ -204,10 +253,34 @@ fit_mmrm <- function(
 }
 
 
-#' Title
+
+#' Fit an mmrm model via multiple optimizers
```
Comment (nociale) [inline R/mmrm.R:257]:
```suggestion
#' Fit an MMRM model via multiple optimizers
```

File: R/mmrm.R [hunk 14]
```diff
@@ -204,10 +253,34 @@ fit_mmrm <- function(
...
+#'
+#'
+#' The function attempts to fit a mmrm model using the optimizer as specified in `optimizer`
```
Comment (nociale) [inline R/mmrm.R:260]:
```suggestion
#' The function attempts to fit a MMRM model using the optimizer as specified in `optimizer`
```

File: R/mmrm.R [hunk 15]
```diff
@@ -204,10 +253,34 @@ fit_mmrm <- function(
...
+#' If `optimizer` is of length > 1 then it will loop through all the values until one of them is able
+#' to converge. That is to say if a fit fails to converge it will move onto the next value of `optimizer` and
+#' try again.
+#'
+#' If `optimizer` is a list then the names of the list with taken to be the required `optimizer`
```
Comment (nociale) [inline R/mmrm.R:265]: "If `optimizer` is a list then the names of the list with taken to be the required `optimizer`" sorry is this correct grammatically? I don't understand what you mean :)

Comment (gowerc) [reply R/mmrm.R:265]: aww sorry "with" should be "will be" i.e.

"names of the list **will be** taken to be the

Comment (gowerc) [reply R/mmrm.R:265]:
```suggestion
#' If `optimizer` is a list then the names of the list will be taken to be the required `optimizer`
```

**Review by nociale — APPROVED** [review]

**Pull request merged**